tutorial-review.md

```diff
@@ -0,0 +1,1053 @@
+
+1. When hitting `/posts/post_id` the client calls the server API to fetch the specific blog post.
+2. The post is stored in the component state.
+3. Using react-boostrap, we create a simple-but-effective UI for the users to read the post.
+4. On top of this, we add 2 buttons to either "edit" or "delete" the posts. Specifically, "edit" is nothing more than a link to `/posts/post_id/edit` and delete calls DELETE `/api/posts/:post_id` and then redirect the user to the homepage.
+
+Open the terminal and create `post.js` under `/pages`:
+
+```bash
+touch post.js
+```
+
+```javascript
+import { useEffect, useState } from 'react';
+import { useParams, useNavigate, Link } from 'react-router-dom';
+import Container from 'react-bootstrap/Container';
+import Button from 'react-bootstrap/Button';
+import http from '../lib/http';
+import formatDate from '../lib/formatDate';
+
+const Post = () => {
+  const { id: postId } = useParams();
+  const [post, setPost] = useState({});
+  const navigate = useNavigate();
+  // Fetch the single blog post
+  useEffect(() => {
+    async function fetchData() {
+      const { data } = await http.get(`/api/posts/${postId}`);
+      setPost(data.data.post);
+    }
+    fetchData();
+  }, [postId]);
+  // Delete the post and redirect the user to the homepage
+  const deletePost = async () => {
+    await http.delete(`/api/posts/${postId}`);
+    navigate('/');
+  }
+  
+  
+  return (
+    <>
+      <Container className="my-5 text-justified" style={{ maxWidth: '800px' }}>
+        <h1>{post.title}</h1>
+        <div className="text-secondary mb-4">{formatDate(post.createdAt)}</div>
+        {post.tags?.map((tag) => <span>{tag} </span>)}
+        <div className="h4 mt-5">{post.content}</div>
+        <div className="text-secondary mb-5">- {post.author}</div>
+        <div className="mb-5">
+          <Link
+            variant="primary"
+            className=" btn btn-primary m-2"
+            to={`/posts/${postId}/edit`}
+          >
+            Edit
+          </Link>
+          <Button variant="danger" onClick={deletePost}>Delete</Button>
+        </div>
+        <Link to="/" style={{ textDecoration: 'none' }}>&#8592; Back to Home</Link>
+      </Container>
+    </>
+  );
+};
+
+export default Post;
+```
+
+The UI will look like this:
+
+[react-post-content]
+
+As we redirect the user to another page when editing the blog post, create the file `edit.js` inside `/pages`:
+
+```bash
+touch edit.js
+```
+
+The UI will show a form filled with the blog post data for title, author, content and tags. Users can
+1. Edit each one of the fields
+2. submit the data to the server by calling PUT `/api/posts/:post_id`
+
+Note that we are using `react-hook-form` to register fields, collect the data and submit to the server. For the purpose of the tutorial we are not performing any validation on the data but it is fairly simple to be added thanks to react-hook-form simple API.
```

Comment:
How and when are we using this? Was it a part of a package? Could you add some explanation around this to clarify its use?

Reply:
Actually the tutorial tells the users to install it at the beginning, under "Create the project client and server repos", around line 214. Basically the only usage is to submit the form to the server.

Reply:
You're right, thanks Sam!